Lib/ufo2ft/filters/transformations.py

```diff
@@ -92,6 +100,19 @@ def set_context(self, font, glyphSet):
 
         origin_height = self.get_origin_height(font, self.options.Origin)
 
+        # Metrics
+        ctx.metrics = dict()
+        if self.options.LSB:
+            ctx.metrics["leftMargin"] = self.options.LSB
+        if self.options.RSB:
+            ctx.metrics["rightMargin"] = self.options.RSB
+        if self.options.TSB:
+            ctx.metrics["topMargin"] = self.options.TSB
+        if self.options.BSB:
+            ctx.metrics["bottomMargin"] = self.options.BSB
+        if self.options.VerticalOrigin:
+            ctx.metrics["verticalOrigin"] = self.options.VerticalOrigin
```

Comment:
why don't you just use the metrics value defined in `self.options`? The context is only for stuff that may vary from glyph to another, but this options should be global for all the glyphs processed by this filter instance.

Reply:
> stuff that may vary from glyph to another

sorry, I meant stuff that may vary from one _font_ to another.

I see, you're mapping the Glyphs.app-specific option names (e.g. LSB, RSB) to defcon Glyph-specific attribute names (e.g. "leftMargin", "rightMargin").
I think this should be moved to `start` method, which is run only once at filter initialization, since it is not dependent on the font you're currenty running the filter onto.